api/generate.go

```diff
@@ -0,0 +1,60 @@
+package main
```

Comment:
we already have some code generation in skv2 for multicluster; maybe this belongs alongside that?

Reply:
We have templates to generate multi cluster reconcilers, but I don't think we have code to specifically generate code in this repo to be consumed by other repos, I may be wrong though.